src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOperation.scala

```diff
@@ -52,11 +52,11 @@ object InstanceUpdateOperation {
     */
   case class LaunchOnReservation(
       instanceId: Instance.Id,
-      newTaskId: Task.Id,
+      newTaskIds: Map[Task.Id, Task.Id], // mapping of old task IDs to new ones
```

Comment:
I had the impression this was a map of instanceId to taskId. Should it be? Having a taskId -> taskId mapping seems odd.

Reply:
It is needed to resident tasks in order to have somewhat stable task IDs, stable here means that only the UUID part changes upon relaunch. It am not crazy about it too, probably it could be done in a better way, but it would mean yet another refactoring which I cannot afford right now. Should I add this to the to-do list?

Reply:
maybe the field then should be named oldToNewTaskIds? (then we can lose the comment too)

Or we can explain in the comment why is the mapping needed in the first place. What do you think?

Reply:
Makes sense. Will do.